src/hyperion/parameters/beamline_parameters.py

```diff
@@ -30,22 +32,55 @@ def from_lines(cls, config_lines: list[str]):
             if len(param) == 2
         ]
         for i, (param, value) in enumerate(config_pairs):
-            if value == "Yes":
-                config_pairs[i] = (config_pairs[i][0], True)
-            elif value == "No":
-                config_pairs[i] = (config_pairs[i][0], False)
-            elif value in BEAMLINE_PARAMETER_KEYWORDS:
-                pass
-            else:
-                config_pairs[i] = (config_pairs[i][0], float(config_pairs[i][1]))
+            try:
+                # BEAMLINE_PARAMETER_KEYWORDS effectively raw string but whitespace removed
+                if value not in BEAMLINE_PARAMETER_KEYWORDS:
+                    config_pairs[i] = (
+                        param,
+                        cls.parse_value(value),
+                    )
+            except Exception as e:
+                LOGGER.warning(f"Unable to parse {file_name} line {i}: {e}")
+
         ob.params = dict(config_pairs)
         return ob
 
     @classmethod
     def from_file(cls, path: str):
         with open(path) as f:
             config_lines = f.readlines()
-        return cls.from_lines(config_lines)
+        return cls.from_lines(path, config_lines)
+
+    @classmethod
+    def parse_value(cls, value: str):
+        if value[0] == "[":
+            return cls.parse_list(value[1:].strip())
+        else:
+            return cls.parse_list_element(value)
+
+    @classmethod
+    def parse_list_element(cls, value: str):
+        if value == "Yes":
+            return True
+        elif value == "No":
+            return False
+        else:
+            return float(value)
+
+    @classmethod
+    def parse_list(cls, value: str):
```

Comment:
```suggestion
    @classmethod
    def parse_list(cls, value: str):
        list_output = []

        if value[-1] != "]":
            raise Exception("Missing closing ']' in list expression")

        remaining = value[:-1]

        for i in remaining.split(","):
            i = i.strip()
            list_output.append(cls.parse_list_element(i))

        return list_output
```
I think this is just personal preference as I always get confused with indices, the logic you have is perfectly fine!

Reply:
I wrote it that way because I originally intended it to tokenise chunks of the remaining string incrementally, so that it could handle nested lists. However it turned out that due to not having sufficiently well defined syntax we don't have enough information to parse it in as nice a way as I wanted without adding a lot of complexity.